Flakey Unit Test: test_json.py - test_arbitrary_json

This [unit test](https://github.com/apple/turicreate/blob/d8aa46a5e79966029796b22ab52616dccfbc2e64/src/python/turicreate/test/test_json.py#L360) should be rewritten or removed. 

Failure example 1:
```
_________________________ JSONTest.test_arbitrary_json _________________________

self = <turicreate.test.test_json.JSONTest testMethod=test_arbitrary_json>

    @hypothesis.settings(derandomize=True, suppress_health_check=[hypothesis.HealthCheck.too_slow])
>   @hypothesis.given(hypothesis_json)
    def test_arbitrary_json(self, json_obj):

test_json.py:359: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../hypothesis/core.py:599: in execute
    % (test.__name__, text_repr[0])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <hypothesis.core.StateForActualGivenExecution object at 0x13401e710>
message = "Hypothesis test_arbitrary_json(self=<turicreate.test.test_json.JSONTest testMethod=test_arbitrary_json>, json_obj=[{'t': 0}]) produces unreliable results: Falsified on the first call but did not on a subsequent one"

    def __flaky(self, message):
        if len(self.falsifying_examples) <= 1:
>           raise Flaky(message)
E           hypothesis.errors.Flaky: Hypothesis test_arbitrary_json(self=<turicreate.test.test_json.JSONTest testMethod=test_arbitrary_json>, json_obj=[{'t': 0}]) produces unreliable results: Falsified on the first call but did not on a subsequent one

../../hypothesis/core.py:769: Flaky
```

Example 2:
```
Falsifying example: test_arbitrary_json(self=<turicreate.test.test_json.JSONTest testMethod=test_arbitrary_json>, json_obj=[{'Y': [{'2AB': -0.0,
    '7h': 13,
    'A': '',
    'CkEm': 0.5,
    'EUUY': -2.220446049250313e-16,
    'JBbEZ': '',
    'LgZcU': '',
    'Y': '',
    'Zvgdo': '',
    'ddDng': '',
    'o1Qj6jrL': ''}]}])
Unreliable test timings! On an initial run, this test took 305.89ms, which exceeded the deadline of 200.00ms, but on a subsequent run it took 11.93 ms, which did not. If you expect this sort of variability in your test timings, consider turning deadlines off for this test by setting deadline=None.
```
Comparing run time across runs doesn't make sense. In many cases the unit test returns early if the random input isn't in the correct format.

--- a/scripts/requirements.txt
+++ b/scripts/requirements.txt
@@ -24,7 +24,6 @@ boto3>=1.12
 
 argparse>=1.2.1
 pytest>=4.6.3
-hypothesis>=4.50,<5.0
 wheel>=0.29.0
 six>=1.12.0
 requests>=2.9.1

--- a/src/python/turicreate/test/test_json.py
+++ b/src/python/turicreate/test/test_json.py
@@ -17,7 +17,6 @@ from __future__ import absolute_import as _
 
 import array
 import datetime
-import hypothesis
 import json  # Python built-in JSON module
 import math
 import os
@@ -69,58 +68,6 @@ _SFrameComparer = util.SFrameComparer()
 
 
 class JSONTest(unittest.TestCase):
-    # Only generate lists of dicts, but allow nearly-arbitrary JSON inside those.
-    # However, limit to length 1 to make sure the keys are the same in all rows.
-
-    # Known bug #1: escaped chars give different behavior in SFrame JSON parsing
-    # vs Python's built-in JSON module. Not sure which is correct:
-    """
-    Original JSON:  [{"": [{"\f": 0}]}]
-    Expected:  +---------------+
-    |       X1      |
-    +---------------+
-    | [{'\x0c': 0}] |
-    +---------------+
-    [1 rows x 1 columns]
-
-    Actual:  +--------------+
-    |      X1      |
-    +--------------+
-    | [{'\\f': 0}] |
-    +--------------+
-    [1 rows x 1 columns]
-    """
-    # In the meantime, let's use `string.ascii_letters + string.digits` instead of
-    # `string.printable` (which contains the problematic characters).
-    hypothesis_json = hypothesis.strategies.lists(
-        hypothesis.strategies.dictionaries(
-            keys=hypothesis.strategies.text(string.ascii_letters + string.digits),
-            values=hypothesis.strategies.recursive(
-                # Known bug #2: [{"": null}] parses as "null" in SFrame, and should be None
-                # Once this is fixed, uncomment the line below.
-                # hypothesis.strategies.none() |
-                # Known bug #3: [{"": false}] parses as "false" in SFrame, and should be 0
-                # Once this is fixed, uncomment the line below.
-                # hypothesis.strategies.booleans() |
-                hypothesis.strategies.integers(
-                    min_value=-(2 ** 53) + 1, max_value=(2 ** 53) - 1
-                )
-                | hypothesis.strategies.floats()
-                | hypothesis.strategies.text(string.ascii_letters + string.digits),
-                lambda children: hypothesis.strategies.lists(children, 1)
-                | hypothesis.strategies.dictionaries(
-                    hypothesis.strategies.text(string.ascii_letters + string.digits),
-                    children,
-                    min_size=1,
-                ),
-            ),
-            min_size=1,
-            max_size=1,
-        ),
-        min_size=1,
-        max_size=1,
-    )
-
     def _assertEqual(self, x, y):
         if type(x) in [long, int]:
             self.assertTrue(type(y) in [long, int])
@@ -369,52 +316,6 @@ class JSONTest(unittest.TestCase):
             sf_actual = SFrame.read_json(f.name, orient="lines")
             sf_expected = SFrame(df)
             _SFrameComparer._assert_sframe_equal(sf_expected, sf_actual)
-
-    @pytest.mark.xfail(
-        reason="Non-deterministic test failure tracked in https://github.com/apple/turicreate/issues/2934"
-    )
-    # deterministic across runs, and may take a while
-    @hypothesis.settings(
-        derandomize=True, suppress_health_check=[hypothesis.HealthCheck.too_slow]
-    )
-    @hypothesis.given(hypothesis_json)
-    def test_arbitrary_json(self, json_obj):
-        # Known bug #1: escaped chars give different behavior in SFrame JSON parsing
-        # vs Python's built-in JSON module. Not sure which is correct.
-        # Workaround captured in definition of `hypothesis_json`
-
-        # Known bug #2: [{"": null}] parses as "null" in SFrame, and should be None
-        # Workaround captured in definition of `hypothesis_json`
-
-        # Known bug #3: [{"": false}] parses as "false" in SFrame, and should be 0
-        # Workaround captured in definition of `hypothesis_json`
-
-        try:
-            json_text = json.dumps(json_obj, allow_nan=False)
-        except:
-            # not actually valid JSON - skip this example
-            return
-
-        try:
-            expected = SFrame(json_obj).unpack("X1", column_name_prefix="")
-        except TypeError:
-            # something like TypeError: A common type cannot be infered from types integer, string.
-            # TC enforces all list items have the same type, which
-            # JSON does not necessarily enforce. Let's skip those examples.
-            return
-        with tempfile.NamedTemporaryFile("w") as f:
-            f.write(json_text)
-            f.flush()
-
-            try:
-                actual = SFrame.read_json(f.name)
-            except TypeError:
-                # something like TypeError: A common type cannot be infered from types integer, string.
-                # TC enforces all list items have the same type, which
-                # JSON does not necessarily enforce. Let's skip those examples.
-                return
-
-            _SFrameComparer._assert_sframe_equal(expected, actual)
 
     def test_true_false_substitutions(self):
         expecteda = [["a", "b", "c"], ["a", "b", "c"]]

